internal/driver/glfw/canvas.go

```diff
@@ -105,8 +105,9 @@ func (c *glCanvas) PixelCoordinateForPosition(pos fyne.Position) (int, int) {
 }
 
 func (c *glCanvas) Resize(size fyne.Size) {
+	pixelsSize := fyne.NewSize(float32(math.Ceil(float64(size.Width))), float32(math.Ceil(float64(size.Height))))
```

Comment:
I think this is more likely `nearestPixelSize` or something shorter... as it's not really a pixel size.

However this maths does not result in that either - it is currently getting a pixel at the coursest level and multiplying by scale, in effect. So on a HighDPI with fractional content it may be too wide/high.
Consider checking out `roundToPixel` in `internal/painter/gl/draw.go` as this will return the closest pixel. Though in this instance we probably want `Ceil` not `Round` as the canvas must always be at least the minimum size, not rounded down.

Reply:
Wouldn't it be logical that all rounding to pixels logic use the same logic (Ceil or Round) throughout the code?

Reply:
It depends on the intent of the code.
To find the closest pixel you want to round to closest. To get the size that is required to fit the content you must round up as the canvas/window should never be smaller than required

Reply:
I went with nearestSize. I tried to follow all the callers and callees path to see if there was a better location, but the canvas size really need to be adjusted as it is accessed in a few place from the painter directly not being passed around. Also their is multiple callers that have different way of coming at the number and again the call to Resize is where everyone gather, so at this point, I think that's the best place for it.

Reply:
I think this isn't going to be the right value though. For lowDPI (1dp=1px) it will work cool, but what about if we're on super-high DPI? Like iOS:

25dp at 4* DPI is 100px, then if the value was 24.6 then this would return 25 - the 100px equivalent, but it should be 24.5, which is 98px. Meaning that the canvas will be 2px bigger than the window.

Reply:
I guess in the other direction where the content is too small may be worse as we'll get a 2px stripe at the bottom of the window.

Reply:
It sounds like you're confusing canvas and window size - this fix is for canvas size only, so I'm not sure I follow the logic.

Windows move in pixel size increments, so surely we want the canvas to match that?